library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -124,6 +124,34 @@
   protected transient long skipCount = 0;
   private transient OperatorContext context;
 
+  /**
+   * The time between sending batch tuples. Default value is 1 hour. A batch
+   * interval could consist of multiple scan intervals
+   */
+  private long batchIntervalMillis = 60 * 60 * 1000; //1 hour
```

Comment:
In this case the use case is processing time batches, for example, an hourly scan through the directory is part of a single batch
@ajaygit158 please add the use cases to the jira